rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -232,6 +244,26 @@ fn set_non_blocking(fd: RawFd) -> Result<()> {
     }
 }
 
+fn create_tun_device() -> Result<()> {
+    let path = Path::new(std::str::from_utf8(TUN_FILE).unwrap());
+
+    if !path.exists() {
+        let parent_dir = path.parent().unwrap();
+        fs::create_dir_all(parent_dir)?;
+        let permissions = fs::Permissions::from_mode(0o751);
+        fs::set_permissions(parent_dir, permissions)?;
+        unsafe {
+            mknod(
+                TUN_FILE.as_ptr() as _,
+                S_IFCHR,
+                makedev(TUN_DEV_MAJOR, TUN_DEV_MINOR),
+            )
+        };
```

Comment:
`mknod` might fail so you need to check if this returns non-zero and bubble up the error like:

```rs
if unsafe{ mknod(..) } != 0 { return Err(get_last_error()) }
```
we can wrap it in its own funciton